File: app/src/main/java/com/maherhanna/cheeta/State.java
package com.maherhanna.cheeta;

import java.util.Objects;

public class State {
    public long allPieces;
    public int enPassantTarget;
    public int blackCastlingRights;
    public int whiteCastlingRights;

    public State(long allPieces,  int enPassantTarget,int blackCastlingRights,
                 int whiteCastlingRights){

        this.allPieces = allPieces;
        this.enPassantTarget = enPassantTarget;
        this.blackCastlingRights = blackCastlingRights;
        this.whiteCastlingRights = whiteCastlingRights;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        State state = (State) o;
        return allPieces == state.allPieces &&
                enPassantTarget == state.enPassantTarget &&
                blackCastlingRights == state.blackCastlingRights &&
                whiteCastlingRights == state.whiteCastlingRights;
    }

    @Override
    public int hashCode() {
        return Objects.hash(allPieces,enPassantTarget, blackCastlingRights, whiteCastlingRights);
    }
}

File: app/core/src/main/java/com/maherhanna/cheeta/core/State.kt
package com.maherhanna.cheeta.core

import java.util.Objects

class State(
    var allPieces: Long, @JvmField var enPassantTarget: Int, @JvmField var blackCastlingRights: Int,
    @JvmField var whiteCastlingRights: Int
) {
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val state = o as State
        return allPieces == state.allPieces && enPassantTarget == state.enPassantTarget && blackCastlingRights == state.blackCastlingRights && whiteCastlingRights == state.whiteCastlingRights
    }

    override fun hashCode(): Int {
        return Objects.hash(allPieces, enPassantTarget, blackCastlingRights, whiteCastlingRights)
    }
}
